src/odemis/model/_vattributes.py

```diff
@@ -990,22 +990,31 @@ def clip(self, val):
         clipped separately by position for positions 0..n-1
         return (same type as val): value clipped
         """
+        if val in self.choices:
+            return val
+
+        def calcDistance(choice):
+            """Calculate euclidean distance. If choice contains non numbers,
+            an infinite distance is returned."""
+            try:
```

Comment:
Modifying an external variable (ie, `ls`) from a local function works... but it's really tricky to read. 
I'd suggest to move the content of the function inside the for loop.
Also, add a comment to the Exception to indicate what it means (and why it's fine to just pass): choice and val cannot be compared. Maybe you could restrict the Exception to `(TypeError, ValueError)`, to make it slightly more obvious.